woocommerce/woocommerce
# Reviews: Respect "Comment author must fill out name and email" setting

The review/comment form on products was not respecting the WP core setting `require_name_email`. If this is false (0) the name and email fields are not required.

This PR creates those 2 fields dynamically and only adds required/* marks if that setting is true. Validation is unchanged (handled by WP).

Fixes #23120

To test, 

1. go to Settings > Discussion > "Comment author must fill out name and email" unchecked.
2. View product whist logged out
3. Confirm review form does not show * asterisks for name + email
